[boost] update to 1.87.0

miyanyan opened the pull request:
- [x] Changes comply with the [maintainer guide](https://github.com/microsoft/vcpkg-docs/blob/main/vcpkg/contributing/maintainer-guide.md).
- [x] SHA512s are updated for each updated download.
- [ ] The "supports" clause reflects platforms that may be fixed by this new version.
- [ ] Any fixed [CI baseline](https://github.com/microsoft/vcpkg/blob/master/scripts/ci.baseline.txt) entries are removed from that file.
- [x] Any patches that are no longer applied are deleted from the port's directory.
- [x] The version database is fixed by rerunning `./vcpkg x-add-version --all` and committing the result.
- [x] Only one version is added to each modified port's versions file.

miyanyan committed:
[boost] update to 1.87.0

miyanyan committed:
[boost-build] remove port-version

miyanyan committed:
[boost] update versions

miyanyan commented:
there's no 1.87.0 release for boost-compatibility, so I keep it as 1.86.0

miyanyan committed:
[boost] add boost-parser

miyanyan committed:
[boost] update versions

Cheney-W commented:
Regarding the `boost-build` port, it requires manually updating the `SHA512` value in the `portfile.cmake`.

Cheney-W commented:
Below ports should do the same thing as `boost-build`.
```
boost-graph-parallel
boost-mpi
boost-property-map-parallel
```
The line 31 of `scripts/boost/generate-ports.ps1` should be removed.

miyanyan committed:
[boost-build] update hash

miyanyan commented:
> Below ports should do the same thing as `boost-build`.
> 
> ```
> boost-graph-parallel
> boost-mpi
> boost-property-map-parallel
> ```
> 
> The line 31 of `scripts/boost/generate-ports.ps1` should be removed.

thanks

miyanyan committed:
Update generate-ports.ps1

miyanyan committed:
[boost] update mpi hash

miyanyan committed:
[boost] update mpi versions

miyanyan commented:
@Cheney-W asio removed some deprecated facilities, see https://www.boost.org/doc/libs/1_87_0/doc/html/boost_asio/history.html, and this caused some ports build failed。

for https://github.com/glynos/cpp-netlib, it has been archived by the owner on Apr 3, 2020, should we still need to support it?

miyanyan committed:
[cpp-netlib] fix asio build error

miyanyan committed:
[websocketpp] fix asio build error

miyanyan committed:
Update use-asio-iocontext.patch

miyanyan committed:
[freeopcua] fix asio build error

miyanyan committed:
update versions

miyanyan committed:
[boost-iostreams] fix zstd build error when static

miyanyan committed:
[boost] update versions

dg0yt commented on ports/boost-iostreams/portfile.cmake:
```diff
@@ -10,6 +10,9 @@ vcpkg_from_github(
 set(FEATURE_OPTIONS "")
 include("${CMAKE_CURRENT_LIST_DIR}/features.cmake")
+if (VCPKG_LIBRARY_LINKAGE STREQUAL "static")
+    list(APPEND FEATURE_OPTIONS "-DBOOST_IOSTREAMS_ZSTD_TARGET=zstd::libzstd_static")
```
```suggestion
    list(APPEND FEATURE_OPTIONS "-DBOOST_IOSTREAMS_ZSTD_TARGET=zstd::libzstd")
```
if it is necessary at all. This target is always created.

miyanyan replied:
thanks, I changed to 
```
if (BOOST_IOSTREAMS_ENABLE_ZSTD)
    list(APPEND FEATURE_OPTIONS "-DBOOST_IOSTREAMS_ZSTD_TARGET=zstd::libzstd")
endif()
```

miyanyan replied:
@dg0yt there's an error when `vcpkg install avro-cpp:x64-windows-static`:
```
CMake Error at D:/installed/x64-windows-static/share/boost_iostreams/boost_iostreams-targets.cmake:60 (set_target_properties):
  The link interface of target "Boost::iostreams" contains:

    zstd::libzstd

  but the target was not found.  Possible reasons include:

    * There is a typo in the target name.
    * A find_package call is missing for an IMPORTED target.
    * An ALIAS target is missing.

Call Stack (most recent call first):
  D:/installed/x64-windows-static/share/boost_iostreams/boost_iostreams-config.cmake:67 (include)
  D:/a/_work/1/s/scripts/buildsystems/vcpkg.cmake:859 (_find_package)
  D:/installed/x64-windows-static/share/boost/BoostConfig.cmake:67 (find_package)
  D:/installed/x64-windows-static/share/boost/BoostConfig.cmake:128 (boostcfg_find_component)
  D:/installed/x64-windows-static/share/boost/vcpkg-cmake-wrapper.cmake:3 (_find_package)
  D:/a/_work/1/s/scripts/buildsystems/vcpkg.cmake:813 (include)
  D:/downloads/tools/cmake-3.30.1-windows/cmake-3.30.1-windows-i386/share/cmake-3.30/Modules/FindBoost.cmake:610 (find_package)
  D:/installed/x64-windows-static/share/boost/vcpkg-cmake-wrapper.cmake:3 (_find_package)
  D:/a/_work/1/s/scripts/buildsystems/vcpkg.cmake:813 (include)
  CMakeLists.txt:77 (find_package)

```
[config-x64-windows-static-out.log](https://github.com/user-attachments/files/18220913/config-x64-windows-static-out.log)

but if I changed to 
```
if (VCPKG_LIBRARY_LINKAGE STREQUAL "static")
    list(APPEND FEATURE_OPTIONS "-DBOOST_IOSTREAMS_ZSTD_TARGET=zstd::libzstd_static")
endif()
```
it works well, seems target zstd::libzstd not work when build static?
[boost_iostreams-targets.txt](https://github.com/user-attachments/files/18220944/boost_iostreams-targets.txt)
[zstdTargets.txt](https://github.com/user-attachments/files/18220945/zstdTargets.txt)

miyanyan committed:
[boost-iostreams] update portfile.cmake

miyanyan committed:
[wt] fix asio build error

miyanyan committed:
update versions

miyanyan committed:
[cpp-netlib] fix asio build error

miyanyan committed:
[cpprestsdk] fix asio build error

miyanyan committed:
update versions

miyanyan committed:
Update fix-asio-error.patch

miyanyan committed:
[websocketpp, cpprestsdk]fix asio build error

miyanyan committed:
update versions

Tradias commented:
Consider removing `supports` limitations from Boost.Cobalt, see https://github.com/boostorg/cobalt/issues/179#issuecomment-2550934734

miyanyan committed:
[restc-cpp] fix asio build error

miyanyan committed:
[boost-cobalt] remove supports

miyanyan committed:
update versions

miyanyan committed:
[tgbot-cpp] fix asio build error

miyanyan committed:
[sleep-discord] fix asio build error

miyanyan committed:
update versions

miyanyan committed:
[functions-framework-cpp] fix asio build error

miyanyan committed:
[hareflow] fix asio build error

miyanyan committed:
update versions

miyanyan committed:
[pcl] fix asio error

miyanyan committed:
[qwtw] fix asio build error

miyanyan committed:
[quickfast] fix asio build error

miyanyan committed:
update versions

miyanyan committed:
[hazelcast-cpp-client] fix asio error

miyanyan committed:
[boost-geometry] add miss boost-crc

miyanyan committed:
update versions

miyanyan committed:
[cpprestsdk] fix asio error

miyanyan committed:
update versions

miyanyan committed:
Merge remote-tracking branch 'upstream/master' into boost-1.87.0

miyanyan committed:
[quickfast] fix asio build error

miyanyan committed:
[websocketpp] fix asio error

miyanyan committed:
update versions

miyanyan committed:
[nghttp2-asio] fix asio build error

miyanyan committed:
update versions

miyanyan committed:
[boost-cobalt] not support uwp

miyanyan committed:
[boost-iostreams] use zstd static target

miyanyan committed:
[hazelcast-cpp-client] fix boost 1.87 error

boost-math require c++14

miyanyan committed:
update versions

miyanyan committed:
[autodock-vina] require c++14

miyanyan committed:
[azure-storage-cpp] fix asio error

miyanyan committed:
[gtsam] boost math require c++14

miyanyan committed:
[libe57] boost math require c++14

miyanyan committed:
update vcpkg.json

miyanyan committed:
[boost-smart-ptr] remove unnecessary dependencies

miyanyan committed:
[vowpal-wabbit] boost math require c++14

miyanyan committed:
update versions

miyanyan committed:
Update baseline.json

miyanyan commented:
boost-regex got some error on x64_osx, why boost-regex not header-only and drop c++03 support? see https://www.boost.org/doc/libs/1_87_0/libs/regex/doc/html/boost_regex/install.html

seems boost-regex only need four dependencies
https://github.com/boostorg/regex/blob/boost-1.87.0/CMakeLists.txt#L20-#L26

miyanyan committed:
[boost-json] add arm32 patch

miyanyan committed:
[boost-regex] remove unnecessaries dependencies

miyanyan committed:
update versions

miyanyan commented:
no idea why boost-graph got error on arm_neon_android, can't find asm code in boost-graph files...
```
Run Build Command(s): /vcpkg/downloads/tools/ninja/1.10.2-linux/ninja -v -v -j33 install
[1/4] /android-ndk-r27c/toolchains/llvm/prebuilt/linux-x86_64/bin/clang++ --target=armv7-none-linux-androideabi21 --sysroot=/android-ndk-r27c/toolchains/llvm/prebuilt/linux-x86_64/sysroot -DBOOST_ATOMIC_NO_LIB -DBOOST_ATOMIC_STATIC_LINK -DBOOST_CHRONO_NO_LIB -DBOOST_CHRONO_STATIC_LINK -DBOOST_CONTAINER_NO_LIB -DBOOST_CONTAINER_STATIC_LINK -DBOOST_DATE_TIME_NO_LIB -DBOOST_DATE_TIME_STATIC_LINK -DBOOST_GRAPH_NO_LIB -DBOOST_GRAPH_STATIC_LINK -DBOOST_RANDOM_NO_LIB -DBOOST_RANDOM_STATIC_LINK -DBOOST_SERIALIZATION_NO_LIB -DBOOST_SERIALIZATION_STATIC_LINK -DBOOST_THREAD_NO_LIB -DBOOST_THREAD_STATIC_LINK -DBOOST_THREAD_USE_LIB -I/mnt/vcpkg-ci/b/boost-graph/src/ost-1.87.0-b8107cfa31.clean/libs/graph/include -isystem /mnt/vcpkg-ci/installed/arm-neon-android/include -g -DANDROID -fdata-sections -ffunction-sections -funwind-tables -fstack-protector-strong -no-canonical-prefixes -D_FORTIFY_SOURCE=2 -march=armv7-a -mthumb -Wformat -Werror=format-security -frtti -fexceptions  -fPIC   -fno-limit-debug-info    -fPIC -fvisibility=hidden -fvisibility-inlines-hidden -pthread -MD -MT libs/graph/CMakeFiles/boost_graph.dir/src/graphml.cpp.o -MF libs/graph/CMakeFiles/boost_graph.dir/src/graphml.cpp.o.d -o libs/graph/CMakeFiles/boost_graph.dir/src/graphml.cpp.o -c /mnt/vcpkg-ci/b/boost-graph/src/ost-1.87.0-b8107cfa31.clean/libs/graph/src/graphml.cpp
FAILED: libs/graph/CMakeFiles/boost_graph.dir/src/graphml.cpp.o 
/android-ndk-r27c/toolchains/llvm/prebuilt/linux-x86_64/bin/clang++ --target=armv7-none-linux-androideabi21 --sysroot=/android-ndk-r27c/toolchains/llvm/prebuilt/linux-x86_64/sysroot -DBOOST_ATOMIC_NO_LIB -DBOOST_ATOMIC_STATIC_LINK -DBOOST_CHRONO_NO_LIB -DBOOST_CHRONO_STATIC_LINK -DBOOST_CONTAINER_NO_LIB -DBOOST_CONTAINER_STATIC_LINK -DBOOST_DATE_TIME_NO_LIB -DBOOST_DATE_TIME_STATIC_LINK -DBOOST_GRAPH_NO_LIB -DBOOST_GRAPH_STATIC_LINK -DBOOST_RANDOM_NO_LIB -DBOOST_RANDOM_STATIC_LINK -DBOOST_SERIALIZATION_NO_LIB -DBOOST_SERIALIZATION_STATIC_LINK -DBOOST_THREAD_NO_LIB -DBOOST_THREAD_STATIC_LINK -DBOOST_THREAD_USE_LIB -I/mnt/vcpkg-ci/b/boost-graph/src/ost-1.87.0-b8107cfa31.clean/libs/graph/include -isystem /mnt/vcpkg-ci/installed/arm-neon-android/include -g -DANDROID -fdata-sections -ffunction-sections -funwind-tables -fstack-protector-strong -no-canonical-prefixes -D_FORTIFY_SOURCE=2 -march=armv7-a -mthumb -Wformat -Werror=format-security -frtti -fexceptions  -fPIC   -fno-limit-debug-info    -fPIC -fvisibility=hidden -fvisibility-inlines-hidden -pthread -MD -MT libs/graph/CMakeFiles/boost_graph.dir/src/graphml.cpp.o -MF libs/graph/CMakeFiles/boost_graph.dir/src/graphml.cpp.o.d -o libs/graph/CMakeFiles/boost_graph.dir/src/graphml.cpp.o -c /mnt/vcpkg-ci/b/boost-graph/src/ost-1.87.0-b8107cfa31.clean/libs/graph/src/graphml.cpp
<inline asm>:1:41: error: expected '%<type>' or "<type>"
    1 | .pushsection ".debug_gdb_scripts", "MS",@progbits,1
      |                                         ^
<inline asm>:318:12: error: .popsection without corresponding .pushsection
  318 | .popsection
      |            ^
2 errors generated.
```

miyanyan commented:
> no idea why boost-graph got error on arm_neon_android, can't find asm code in boost-graph files...
> 
> ```
> Run Build Command(s): /vcpkg/downloads/tools/ninja/1.10.2-linux/ninja -v -v -j33 install
> [1/4] /android-ndk-r27c/toolchains/llvm/prebuilt/linux-x86_64/bin/clang++ --target=armv7-none-linux-androideabi21 --sysroot=/android-ndk-r27c/toolchains/llvm/prebuilt/linux-x86_64/sysroot -DBOOST_ATOMIC_NO_LIB -DBOOST_ATOMIC_STATIC_LINK -DBOOST_CHRONO_NO_LIB -DBOOST_CHRONO_STATIC_LINK -DBOOST_CONTAINER_NO_LIB -DBOOST_CONTAINER_STATIC_LINK -DBOOST_DATE_TIME_NO_LIB -DBOOST_DATE_TIME_STATIC_LINK -DBOOST_GRAPH_NO_LIB -DBOOST_GRAPH_STATIC_LINK -DBOOST_RANDOM_NO_LIB -DBOOST_RANDOM_STATIC_LINK -DBOOST_SERIALIZATION_NO_LIB -DBOOST_SERIALIZATION_STATIC_LINK -DBOOST_THREAD_NO_LIB -DBOOST_THREAD_STATIC_LINK -DBOOST_THREAD_USE_LIB -I/mnt/vcpkg-ci/b/boost-graph/src/ost-1.87.0-b8107cfa31.clean/libs/graph/include -isystem /mnt/vcpkg-ci/installed/arm-neon-android/include -g -DANDROID -fdata-sections -ffunction-sections -funwind-tables -fstack-protector-strong -no-canonical-prefixes -D_FORTIFY_SOURCE=2 -march=armv7-a -mthumb -Wformat -Werror=format-security -frtti -fexceptions  -fPIC   -fno-limit-debug-info    -fPIC -fvisibility=hidden -fvisibility-inlines-hidden -pthread -MD -MT libs/graph/CMakeFiles/boost_graph.dir/src/graphml.cpp.o -MF libs/graph/CMakeFiles/boost_graph.dir/src/graphml.cpp.o.d -o libs/graph/CMakeFiles/boost_graph.dir/src/graphml.cpp.o -c /mnt/vcpkg-ci/b/boost-graph/src/ost-1.87.0-b8107cfa31.clean/libs/graph/src/graphml.cpp
> FAILED: libs/graph/CMakeFiles/boost_graph.dir/src/graphml.cpp.o 
> /android-ndk-r27c/toolchains/llvm/prebuilt/linux-x86_64/bin/clang++ --target=armv7-none-linux-androideabi21 --sysroot=/android-ndk-r27c/toolchains/llvm/prebuilt/linux-x86_64/sysroot -DBOOST_ATOMIC_NO_LIB -DBOOST_ATOMIC_STATIC_LINK -DBOOST_CHRONO_NO_LIB -DBOOST_CHRONO_STATIC_LINK -DBOOST_CONTAINER_NO_LIB -DBOOST_CONTAINER_STATIC_LINK -DBOOST_DATE_TIME_NO_LIB -DBOOST_DATE_TIME_STATIC_LINK -DBOOST_GRAPH_NO_LIB -DBOOST_GRAPH_STATIC_LINK -DBOOST_RANDOM_NO_LIB -DBOOST_RANDOM_STATIC_LINK -DBOOST_SERIALIZATION_NO_LIB -DBOOST_SERIALIZATION_STATIC_LINK -DBOOST_THREAD_NO_LIB -DBOOST_THREAD_STATIC_LINK -DBOOST_THREAD_USE_LIB -I/mnt/vcpkg-ci/b/boost-graph/src/ost-1.87.0-b8107cfa31.clean/libs/graph/include -isystem /mnt/vcpkg-ci/installed/arm-neon-android/include -g -DANDROID -fdata-sections -ffunction-sections -funwind-tables -fstack-protector-strong -no-canonical-prefixes -D_FORTIFY_SOURCE=2 -march=armv7-a -mthumb -Wformat -Werror=format-security -frtti -fexceptions  -fPIC   -fno-limit-debug-info    -fPIC -fvisibility=hidden -fvisibility-inlines-hidden -pthread -MD -MT libs/graph/CMakeFiles/boost_graph.dir/src/graphml.cpp.o -MF libs/graph/CMakeFiles/boost_graph.dir/src/graphml.cpp.o.d -o libs/graph/CMakeFiles/boost_graph.dir/src/graphml.cpp.o -c /mnt/vcpkg-ci/b/boost-graph/src/ost-1.87.0-b8107cfa31.clean/libs/graph/src/graphml.cpp
> <inline asm>:1:41: error: expected '%<type>' or "<type>"
>     1 | .pushsection ".debug_gdb_scripts", "MS",@progbits,1
>       |                                         ^
> <inline asm>:318:12: error: .popsection without corresponding .pushsection
>   318 | .popsection
>       |            ^
> 2 errors generated.
> ```

maybe something like https://github.com/ned14/outcome/pull/308

miyanyan committed:
[boost-interprocess] arm32 patch

miyanyan committed:
Merge remote-tracking branch 'upstream/master' into boost-1.87.0

miyanyan committed:
[boost-unordered] arm32 patch

miyanyan committed:
update versions

miyanyan committed:
[boost-math] require c++14

miyanyan committed:
[plustache] set c++ standard to 11

miyanyan committed:
update versions

johnwason committed:
Add patch for robotraconteur to fix boost 1.87.0 build errors

miyanyan committed:
[robotraconteur] update vcpkg.json

miyanyan committed:
update versions

miyanyan commented:
/azp run

miyanyan commented:
@Cheney-W @dg0yt how to rerun the ci, arm64_osx seems stucked...

dg0yt commented:
~~~
git commit --allow-empty -m CI
git push

miyanyan commented:
> ```
> git commit --allow-empty -m CI
> git push
> ```

thanks

miyanyan commented:
mainly changes:
* asio: removed some deprecated facilities, https://www.boost.org/doc/libs/1_87_0/doc/html/boost_asio/history.html, like remove io_service, remove context::work, remove query, remove buffer_cast, remove rfc2818_verification...
* boost-math: require c++14
* boost-json, boost-unordered, boost-interprocess: asm code not work on arm32

JavierMatosD commented:
> asio: removed some deprecated facilities, https://www.boost.org/doc/libs/1_87_0/doc/html/boost_asio/history.html, like remove io_service, remove context::work, remove query, remove buffer_cast, remove rfc2818_verification...

The sheer amount of patching required to force compatibility with `boost-asio` is concerning.  It looks like the following ports are blocking this update:

-  `wt`
-  `websocketpp`
- `tgbot-cpp`
- `sleepy-discord` 
- `robotraconteur`
- `restc-cpp`
- `quickfast`
- `nghttp2-asio`
- `cpp-netlib`

I'm going to bring this up to the team to see how we can proceed.

miyanyan commented:
> > asio: removed some deprecated facilities, https://www.boost.org/doc/libs/1_87_0/doc/html/boost_asio/history.html, like remove io_service, remove context::work, remove query, remove buffer_cast, remove rfc2818_verification...
> 
> The sheer amount of patching required to force compatibility with `boost-asio` is concerning. It looks like the following ports are blocking this update:
> 
> * `wt`
> * `websocketpp`
> * `tgbot-cpp`
> * `sleepy-discord`
> * `robotraconteur`
> * `restc-cpp`
> * `quickfast`
> * `nghttp2-asio`
> * `cpp-netlib`
> 
> I'm going to bring this up to the team to see how we can proceed.

Can we bring the asio deprecated facilities back, like
```
+namespace boost {
+  namespace asio {
+    #if BOOST_VERSION >= 108700
+      using io_service = boost::asio::io_context;
+      using const_buffers_1 = boost::asio::const_buffer;
+      using mutable_buffers_1 = boost::asio::mutable_buffer;
+      using io_context_work = boost::asio::executor_work_guard<boost::asio::io_context::executor_type>;
+      /// Cast a non-modifiable buffer to a specified pointer to POD type.
+      template <typename PointerToPodType>
+      inline PointerToPodType buffer_cast(const mutable_buffer& b) noexcept
+      {
+        return static_cast<PointerToPodType>(b.data());
+      }
+
+      /// Cast a non-modifiable buffer to a specified pointer to POD type.
+      template <typename PointerToPodType>
+      inline PointerToPodType buffer_cast(const const_buffer& b) noexcept
+      {
+        return static_cast<PointerToPodType>(b.data());
+      }
+
+      template <typename Protocol, typename Executor, typename Iterator,
+          BOOST_ASIO_COMPLETION_TOKEN_FOR(void (boost::system::error_code,
+            Iterator)) IteratorConnectToken = default_completion_token_t<Executor>>
+      inline auto async_connect(basic_socket<Protocol, Executor>& s, Iterator begin,
+          IteratorConnectToken&& token = default_completion_token_t<Executor>(),
+          constraint_t<
+            !is_endpoint_sequence<Iterator>::value
+          > = 0,
+          constraint_t<
+            !is_same<Iterator, decay_t<IteratorConnectToken>>::value
+          > = 0,
+          constraint_t<
+            !is_connect_condition<IteratorConnectToken, Iterator>::value
+          > = 0)
+        -> decltype(
+          async_initiate<IteratorConnectToken,
+            void (boost::system::error_code, Iterator)>(
+              declval<detail::initiate_async_iterator_connect<Protocol, Executor>>(),
+              token, begin, Iterator(),
+              declval<detail::default_connect_condition>()))
+      {
+        return async_initiate<IteratorConnectToken,
+          void (boost::system::error_code, Iterator)>(
+            detail::initiate_async_iterator_connect<Protocol, Executor>(s),
+            token, begin, Iterator(), detail::default_connect_condition());
+      }
+    #else
+      using io_context_work = boost::asio::io_service::work;
+    #endif
+  }
+}
+
```

By the way, similar issues may arise with new releases of the standalone Asio library as well

JavierMatosD commented:
@miyanyan, I discussed this with the team. At the very least, the upstreams for all the ports that require patching for compatibility with `boost-asio` need to be notified. According to vcpkg policy, they should be given 30 days to respond. Once their time is up we can look at each port to determine whether we should patch it or delist the port. 

> Can we bring the asio deprecated facilities back

Unfortunately, we cannot.

miyanyan commented:
@JavierMatosD  these ports had not be notified to upstream, let me create a pr for them
| port | last update date | 
|---|---|
| cpp-netlib | 7 years ago(Public archive) |
| websocketpp | 5 years ago |
| freeopcua | 2 years ago |
| wt | 15 hours ago |
| cpprestsdk | 2 months ago | 
| restc-cpp | 4 months ago |
| tgbot-cpp | 3 months ago |
| sleepy-discord | 3 years ago |
| functions-framework-cpp | 2 months ago |
| hareflow | last week |
| qwtw | 7 months ago |
| quickfast | 8 years ago |
| hazelcast-cpp-client | last month |
| nghttp2-asio |  3 years ago |
| azure-storage-cpp | 10 months ago |

miyanyan committed:
Merge remote-tracking branch 'upstream/master' into boost-1.87.0

miyanyan committed:
update versions

yurybura commented:
@JavierMatosD It's been over 30 days now. Can you unlock the Boost library update?

BillyONeal committed:
Merge remote-tracking branch 'origin/master' into HEAD

# Conflicts:
#	ports/restc-cpp/portfile.cmake
#	ports/restc-cpp/vcpkg.json
#	ports/sleepy-discord/vcpkg.json
#	ports/wt/portfile.cmake
#	ports/wt/vcpkg.json
#	versions/baseline.json
#	versions/p-/pcl.json
#	versions/r-/restc-cpp.json
#	versions/s-/sleepy-discord.json
#	versions/w-/wt.json

miyanyan commented:
> I reverted changes to sleepy-discord and wt as they appear to already have fixed this upstream.

Thank you. Following the completion of https://github.com/microsoft/vcpkg/pull/43935, all upstream changes are finalized.

miyanyan committed:
[restc-cpp] remove asio patch

miyanyan committed:
[tgbot-cpp] use upstream patch

miyanyan committed:
update versions

miyanyan commented:
btw, t think there maybe some header-only ports that maybe use deprecated asio functions, so I create this issue https://github.com/microsoft/vcpkg/issues/43348

Hackman1993 commented:
May I Ask why this pr is still pending?

BillyONeal commented:
> May I Ask why this pr is still pending?

Boost.ASIO removed a bunch of backwards compat support that a large chunk of the registry was relying on, and following up those removals with everyone downstream affected takes time.

JavierMatosD commented:
@miyanyan, thank you for the work. I'm going to bring this up to the team to decide what to do with each port https://github.com/microsoft/vcpkg/pull/42678#issuecomment-2597282708

JavierMatosD commented:
@ras0219-msft, @AugP, @BillyONeal, and I discussed this today and reached the following outcomes:
 
cpp-netlib: Delist.
websocketpp: Delist.
freeopcua: Delist.
quickfast: Delist.
hazelcast-cpp-client: Delist. 
nghttp2-asio: Delist.
 
These delistings imply the following delistings as well:
* autobahn
* socket-io-client
* bitserializer
* etcd-cpp-apiv3
* microsoft-signalr
* signalrclient
 
tgbot-cpp: They just did a new release 2 days ago, the existing Boost update PR uses the patch upstream already merged so I think we can merge that.
functions-framework-cpp: The patch is so tiny I think we should just take it.
hareflow: The patch is so tiny I think we should just take it.
qwtw: The patch is so tiny and straightforward I think we should just take it.
cpprestsdk: We can't delist due to high user counts.
azure-storage-cpp: We can't delist due to high user counts.

@BillyONeal 
All of these are deprecated or dead.

@ras0219-msft
- we can remove websocket feature in cpprestsdk and change supports for cpprestsdk to avoid the boost dependencies (windows only)

Delisting websocketcpp will break the "websocket" feature in cpprestsdk. That feature should be removed and cpprestsdk's supports expression should be modified to not trigger boost dependencies which should resolve to windows only.

miyanyan committed:
Merge remote-tracking branch 'upstream/master' into boost-1.87.0

miyanyan committed:
delist some ports that depend on asio

cpp-netlib: Delist.
websocketpp: Delist.
freeopcua: Delist.
quickfast: Delist.
hazelcast-cpp-client: Delist.
nghttp2-asio: Delist.
These delistings imply the following delistings as well:

autobahn

socket-io-client

bitserializer

etcd-cpp-apiv3

microsoft-signalr

signalrclient

miyanyan committed:
[cpprestsdl] update versions

PavelKisliak commented:
Why BitSerializer port was delisted? It's supported and maintained.

miyanyan commented:
> Why BitSerializer port was delisted? It's supported and maintained.

@JavierMatosD seems BitSerializer doesn't depend on websocketpp

BillyONeal commented:
> I'd reconsider the WebSocket++ removal, since this breaks many (including our) VCPKG setup

Then someone needs to actually maintain WebSocket++. This is not vcpkg's choice: We can't leave Boost out of date forever, and we can't become maintainers for every downstream project that is unmaintained.

BillyONeal commented:
> WebSocket++ recipe had an option to not link in any dependencies.

I don't see an obvious way to remove the boost dependency in:

https://github.com/zaphoyd/websocketpp/blob/1b11fd301531e6df35a6107c1e8665b1e77a2d8e/CMakeLists.txt#L176-L245

We have no objection to bringing it back if it can be made to compile. But relying on networking bits which have been unmaintained for 5+ years is probably a bad idea either way.

BillyONeal commented:
> The problem with this unilateral decision is that cpprestsdk relies on websocketpp to provide WebSocket functionality.

It isn't 'unilateral'. A lot more vcpkg customers use boost than use cpprestsdk. Integrity of the registry is more important than any individual thing in the registry.

> cpprestsdk is used by a LOT of vcpkg users.

cpprestsdk is effectively unmaintained, and the websockets part is built on websocketspp which has been unmaintained for even longer. vcpkg can't effectively become the new upstream when one of the ~2500 things in the registry decides they don't want to do it any more.

Nick7K commented:
> > The problem with this unilateral decision is that cpprestsdk relies on websocketpp to provide WebSocket functionality.
> 
> It isn't 'unilateral'. A lot more vcpkg customers use boost than use cpprestsdk. Integrity of the registry is more important than any individual thing in the registry.
> 
> > cpprestsdk is used by a LOT of vcpkg users.
> 
> cpprestsdk is effectively unmaintained, and the websockets part is built on websocketspp which has been unmaintained for even longer. vcpkg can't effectively become the new upstream when one of the ~2500 things in the registry decides they don't want to do it any more.

Fair enough.

BillyONeal commented:
> I do see websocketpp is being updated - https://github.com/zaphoyd/websocketpp/pull/1164

Wow, indeed, back from the dead. We would of course love to put this back if it can compile with current Boost.

reitowo commented:
Why websocketpp is removed from vcpkg?

BillyONeal commented:
> Oh, ok. I just want to say I was using it in manifest mode, after updating baseline in vcpkg.json, it just throw a git error that says invalid tree object or something, not very clear, spent some time debugging it.

Improving the error message is a reasonable issue to report if you have a repro

reitowo commented:
Here's a repro: #45778
